### PR TITLE
feat(kit): add TUI_COUNTRIES_MASKS token

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes https://github.com/Tinkoff/taiga-ui/issues/1226

## What is the new behavior?

Support override countries masks

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/kit/components/input-phone-international/tokens/mask-after-code.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import {InjectionToken} from '@angular/core';
+
+import {MASK_AFTER_CODE_REGEXP} from '../const/countries';
+
+export const TUI_MASK_AFTER_CODE: InjectionToken<RegExp> = new InjectionToken<RegExp>(
```
Comment (waterplea) [inline projects/kit/components/input-phone-international/tokens/mask-after-code.ts:5]: What is the usecase for this token?

File: projects/kit/components/input-phone-international/tokens/countries-masks.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * Use token `TUI_COUNTRIES_MASKS`
+ * TODO: remove in 3.0
+ */
+export const COUNTRIES_MASKS: PhoneMaskByCountry = Object.entries<Country>(
+    COUNTRIES,
+).reduce<PhoneMaskByCountry>(countriesMasksReducer, {} as PhoneMaskByCountry);
+
+export const TUI_COUNTRIES_MASKS: InjectionToken<PhoneMaskByCountry> =
+    new InjectionToken<PhoneMaskByCountry>('Countries masks', {
+        factory: () => COUNTRIES_MASKS,
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/kit/components/input-phone-international/tokens/countries-masks.ts:8]: It's done for backwards compatibility. Please, tell me what should I do here?

File: projects/kit/components/input-phone-international/tokens/countries-masks.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+/**
+ * @deprecated will be deleted soon
+ * Use token `TUI_COUNTRIES_MASKS`
+ * TODO: remove in 3.0
+ */
+export const COUNTRIES_MASKS: PhoneMaskByCountry = Object.entries<Country>(
+    COUNTRIES,
+).reduce<PhoneMaskByCountry>(countriesMasksReducer, {} as PhoneMaskByCountry);
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/kit/components/input-phone-international/tokens/countries-masks.ts]: image

[the remark this answers is not in the document]
Comment (splincode) [reply projects/kit/components/input-phone-international/tokens/countries-masks.ts]: fixed

File: projects/kit/components/input-phone-international/tokens/mask-after-code.ts (hunk 1 above)
Comment (splincode) [reply projects/kit/components/input-phone-international/tokens/mask-after-code.ts]: fixed

File: projects/kit/components/input-phone-international/interfaces/phone-mask-by-country.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import {TuiCountryIsoCode} from '@taiga-ui/i18n';
+
+export type PhoneMaskByCountry = Record<TuiCountryIsoCode, string>;
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/kit/components/input-phone-international/interfaces/phone-mask-by-country.ts]: fixed

File: projects/kit/components/input-phone-international/tokens/countries-masks.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (waterplea) [reply projects/kit/components/input-phone-international/tokens/countries-masks.ts:8]: Looks like I misread the code, sorry.

**Review by waterplea — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**